addons/website/models/res_users.py

```diff
@@ -87,3 +87,12 @@ def authenticate(cls, db, login, password, user_agent_env):
                 visitor_pre_authenticate_sudo.access_token = user_partner.id
                 visitor_pre_authenticate_sudo._update_visitor_last_visit()
         return uid
+
+    def write(self, vals):
+        # If the user was a portal and becomes internal, remove website_id.
+        portal_users = self.filtered(lambda user: user._is_portal())
+        ret = super().write(vals)
+        users_became_internal = portal_users.filtered(lambda user: user._is_internal())
+        if users_became_internal:
+            users_became_internal.write({'website_id': False})
```

Comment:
```suggestion
            users_became_internal.website_id = False
```
No need of the `write()` notation for a single field. It works for recordset on relational fields :+1: